File: frontend/src/types/item.ts
export interface ItemInfo {
  item_id: string;
  title: string;
  stock: number;
}

export interface ItemOptionChoice {
  option_variation_id: string;
  variation_name: string;
  price: number;
}

export interface ItemOption {
  option_id: string;
  option_name: string;
  choices: ItemOptionChoice[];
}

export interface ItemVariation {
  variation_id: string;
  variation_name: string;
  stock: number;
}

File: backend/app/schemas/item.py
from pydantic import BaseModel, Field


class ItemRead(BaseModel):
    item_id: str
    title: str
    stock: int


class ItemDescriptionUpdate(BaseModel):
    detail: str = Field(min_length=1)


class ItemUpdateRead(BaseModel):
    item_id: str
    success: bool


class ItemOptionChoiceRead(BaseModel):
    option_variation_id: str
    variation_name: str
    price: int


class ItemOptionRead(BaseModel):
    option_id: str
    option_name: str
    choices: list[ItemOptionChoiceRead]


class ItemVariationRead(BaseModel):
    variation_id: str
    variation_name: str
    stock: int
